private/protocol/rest/build.go

```diff
@@ -173,7 +173,7 @@ func buildHeader(header *http.Header, v reflect.Value, name string, tag reflect.
 	return nil
 }
 
-func buildHeaderMap(header *http.Header, v reflect.Value, tag reflect.StructTag) error {
+func buildHeaderMap(header *http.Header, v reflect.Value, tag reflect.StructTag, normalize bool) error {
```

Comment:
Does the SDK's marshaling of header maps need to be symmetrical with unmarshaling into an map[string]string? The HTTP header key case in the HTTP message isn't directly tied to how the unmarshaled API parameter represented. If we separate these two concepts, the HTTP headers would stay canonical, and the SDK's `headers` API shape type would be lowercase keys.  The wire representation of these keys would be decoupled.